### PR TITLE
Documentation upgrades

### PR DESCRIPTION
**Pull request by simonwo**
- Add business rules to the documentation.
- Fix up documentation to not emit so many warnings
    - Updates to Pygments that are [about to land](https://github.com/pygments/pygments/pull/1657) will
      take care of the remaining issue with `dot`.

### CODE REVIEW

File: importer/handlers.py [hunk 1]
```diff
@@ -79,16 +79,19 @@ class BaseHandler(metaclass=BaseHandlerMeta):
     This effectively takes place in 8 stages:
 
     Init:
+
     1) The handler is initialised with the initial data.
```
Comment (NDevox) [inline importer/handlers.py:83]: If we `)` -> `.` the above, should we not do that for these?

**Review by NDevox — APPROVED** [review]
Yep looks good to me. One minor comment about consistency, but not hugely important.

File: importer/handlers.py (hunk 1 above)
Comment (simonwo) [reply importer/handlers.py:83]: I think it turns out that the `)` -> `.` is not actually meaningful – initially tried this as a fix when I hadn't read the docs fully.

Comment (simonwo) [reply importer/handlers.py:83]: So perhaps should change it back, but I'll leave that to be done when we update Pygments.

**Pull request merged**